crates/cxx-qt-gen/src/writer/cpp/source.rs

```diff
@@ -49,36 +43,12 @@ fn qobjects_source(generated: &GeneratedCppBlocks) -> Vec<String> {
 
             {methods}
             {namespace_end}
-
-            namespace {namespace_internals} {{
-            ::std::unique_ptr<{ident}>
-            newCppObject()
-            {{
-              return ::std::make_unique<{ident}>();
-            }}
-            }} // namespace {namespace_internals}
         "#,
         ident = qobject.ident,
         namespace_start = namespace_start,
         namespace_end = namespace_end,
-        namespace_internals = qobject.namespace_internals,
-        base_class = qobject.base_class,
         rust_ident = qobject.rust_ident,
-        methods = qobject.blocks.methods.iter().filter_map(pair_as_source).collect::<Vec<String>>().join("\n"),
-        members = {
-            let mut members = vec![
-                format!(", m_rustObj({namespace_internals}::createRs())", namespace_internals = qobject.namespace_internals),
-            ];
-
-            if qobject.locking {
-                members.extend(vec![
-                    ", m_rustObjMutex(::std::make_shared<::std::recursive_mutex>())".to_string(),
-                ]);
-            }
-
-            members.extend(qobject.blocks.members.iter().filter_map(pair_as_source).collect::<Vec<String>>());
-            members.join("\n  ")
-        },
+        methods = qobject.blocks.methods.iter().chain(qobject.blocks.private_methods.iter()).filter_map(pair_as_source).collect::<Vec<String>>().join("\n"),
         deconstructors = qobject.blocks.deconstructors.join("\n  "),
```

Comment:
i wonder if in a further commit deconstructors should be handled in a similar way and built up to a normal public method in the `generator` phase rather than happening in the `writer` phase.

Especially if we are going to have a trait in the future to allow devs to run code in the C++ context in the deconstructor. But maybe this isn't required.

Anyway a separate future change i guess?

Reply:
yes, I think this will be a change we'll make at some point. But it can wait for now.